lib/xhr-mock-proxy-handler.js

```diff
@@ -0,0 +1,50 @@
+var window              = require('global');
+var Promise             = require('lie');
+var realXMLHttpRequest  = window.XMLHttpRequest;
+
+function parseXhrHeaders(xhr) {
+    var headers = {};
+    var rawHeaders = xhr.getAllResponseHeaders().split('\r\n');
+
+    for (var i = 0; i < rawHeaders.length; i++) {
+        var rawHeader = rawHeaders[i].trim();
+                
+        if (rawHeader !== '') {
+            var splittedHeader = rawHeader.split(": ");
+            var headerName = splittedHeader[0];
+            var headerValue = splittedHeader[1];
+            headers[headerName] = headerValue;
+        }
+    }
+
+    return headers;
+}
+
+module.exports = function(req, res) {
+    return new Promise(function(resolve, reject) {
+        var xhr = new realXMLHttpRequest();
```

Comment:
How portable is directly relying on XMLHttpRequest? Isn't there a reason there's so many wrappers (browser support)?